This patch removes the -Wdocumentation flag from the CMake build.
See https://github.com/llvm/llvm-project/issues/68524 and
https://github.com/apple/swift/issues/69154 for more details.
--- swift/cmake/modules/SwiftSharedCMakeConfig.cmake
+++ swift/cmake/modules/SwiftSharedCMakeConfig.cmake
@@ -323,9 +323,6 @@ macro(swift_common_cxx_warnings)
     append_if(CXX_SUPPORTS_WE4062 "/we4062" CMAKE_CXX_FLAGS)
   endif()
 
-  check_cxx_compiler_flag("-Werror -Wdocumentation" CXX_SUPPORTS_DOCUMENTATION_FLAG)
-  append_if(CXX_SUPPORTS_DOCUMENTATION_FLAG "-Wdocumentation" CMAKE_CXX_FLAGS)
-
   check_cxx_compiler_flag("-Werror -Wimplicit-fallthrough" CXX_SUPPORTS_IMPLICIT_FALLTHROUGH_FLAG)
   append_if(CXX_SUPPORTS_IMPLICIT_FALLTHROUGH_FLAG "-Wimplicit-fallthrough" CMAKE_CXX_FLAGS)
